src/ProfSpline/ObservableI.h

```diff
@@ -0,0 +1,25 @@
+// Class to handle all observables as a whole
+// This is a interface class
+
+#ifndef _OBSERVABLE_I_
+#define _OBSERVABLE_I_
+#include "Framework/EventGen/EventRecord.h"
+#include <string>
+#include <vector>
+namespace genie {
+namespace rew {
+class ObservableI {
+public:
+  // virtual void Configure(/*some way of configurating*/) = 0;
+
+  // calculate the value for each observable
+  // maybe we can merge with binning lookup function below
+  // but seperating them may benefit the idea of merging implementation
+  // of corresponding ObservablePrediction in comparison package?
+  virtual std::vector<double> HandleEventRecord(const EventRecord *event) = 0;
```

Comment:
I think I like this class. It matched the Professor Concept naming. 

HandleEventRecord is not the best name. We can be more specific. Something like GetKinematicVariables. I also think we can define a structure to move around this information: instead of returning a vector we can return a generic KinematicVariable, in this way the Observable matches the Comparisons concepts and we return an object which matches the Reweight and Generator environment.

Reply:
Pure coding style: pass a 
```C++
const EventRecord &
```
Let's try to avoid unnecessary pointers